sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -337,6 +337,10 @@ trait CheckAnalysis extends PredicateHelper {
 
           case Tail(limitExpr, _) => checkLimitLikeClause("tail", limitExpr)
 
+          case Union(_, byName, allowMissingCol) if byName || allowMissingCol =>
+            failAnalysis("Union should not be with true `byName` or " +
```

Comment:
To be safe, shall we override `Union.resolved` to include this?

Reply:
I add `!(byName || allowMissingCol)` to the condition at `Union.resolved` now. Then do we still need to check them at `CheckAnalysis` like above? Or just follow other nodes?

Reply:
Since this is not something users can hit(only bug can trigger this), maybe we can just remove this check here.